### PR TITLE
SONARXML-41 Fix FP on S1778 with BOM

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java [hunk 1]
```diff
@@ -95,10 +97,11 @@ private void checkForCharactersBeforeProlog(FileSystem fileSystem) {
     }
   }
 
-  private void checkBOM(String line) {
-    if (line.startsWith(BOM_CHAR)) {
-      characterDeltaForHighlight = -1;
+  private boolean isFileBeginning(int line, int column, boolean hasBOM) {
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java]: MAJOR Make "isFileBeginning" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java [hunk 2]
```diff
@@ -95,10 +97,11 @@ private void checkForCharactersBeforeProlog(FileSystem fileSystem) {
     }
   }
 
-  private void checkBOM(String line) {
-    if (line.startsWith(BOM_CHAR)) {
-      characterDeltaForHighlight = -1;
+  private static boolean isFileBeginning(int line, int column, boolean hasBOM) {
```
Comment (vilchik-elena) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java]: @pynicolas Are you sure about this method name? As I understand, it does the opposite

Comment (pynicolas) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlFile.java]: @vilchik-elena sure! I'm going to fix that.

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java [hunk 3]
```diff
@@ -156,8 +156,8 @@ private void highlightStartElement(XMLStreamReader xmlReader, int startOffset) {
   }
 
   private void highlightXmlDeclaration() {
-    if (content.startsWith(XML_DECLARATION_TAG)) {
-      int startOffset = 0;
+    int startOffset = content.startsWith(XmlFile.BOM_CHAR) ? 1 : 0;
+    if (content.startsWith(XML_DECLARATION_TAG, startOffset)) {
```
Comment (vilchik-elena) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java]: @pynicolas I can't understand why this change is required (as "content" should be the same as before this commit)

Comment (vilchik-elena) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java]: @pynicolas forget it, I understood

**Pull request merged**